File: 01_Language/01_Functions/python/preg_match.py
# coding: utf-8

import re


def preg_match(pattern, subject):
    p = re.compile(pattern)
    return p.search(subject) is not None


if __name__ == '__main__':
    print(preg_match('php', 'PHP is the web scripting language of choice.'))

    p = re.compile('php', re.I)
    print(p.search('PHP is the web scripting language of choice.'))

File: 01_Language/01_Functions/java/src/main/java/com/onevgo/functions/PregMatch.java
package com.onevgo.functions;

import java.util.regex.Pattern;

public class PregMatch {
    public static boolean pregMatch(String pattern, String subject) {
        Pattern compile = Pattern.compile(pattern);
        return compile.matcher(subject).find();
    }

    public static boolean pregMatch(Pattern pattern, String subject) {
        return pattern.matcher(subject).find();
    }

    public static void main(String[] args) {
        Pattern pattern = Pattern.compile("php", Pattern.CASE_INSENSITIVE);
        System.out.println(pregMatch(pattern, "PHP is the web scripting language of choice."));
    }
}
